Add/charge guide (#695)
zeit/docs: pages/guides/deploying-charge-with-now.mdx

<Note>
  Charge automatically imports React into <InlineCode>jsx</InlineCode> files so
  you don't have to.
</Note>

Let's create a simple `layout.html.jsx` file inside the `/src` directory with the following content:

```jsx
export default ({ children }) => {
  return children
}
```

<Caption>
  An example <InlineCode>layout.html.jsx</InlineCode> file in a Charge project.
</Caption>

If at any point we want to see how our changes look, we can serve them locally with the script we added earlier:

<TerminalInput>yarn start</TerminalInput>
<Caption>
  Starting up a local development server using a script.
</Caption>

Charge will open a new tab for you, if not you can navigate to `localhost:2468` to see the changes you make, live-reloading is enabled through [Browsersync](https://browsersync.io/).

Now that we have a layout component, let's use it to wrap an `.mdx` file. Create a file named `about.html.mdx` inside your `/pages` directory.

Let's add some content to our about page and export it inside of the layout component we created, your `about.html.mdx` should look similar to this:

```markdown
import Layout from '../layout.html.jsx'

> Welcome to my new Charge site!

Reviewer: ```suggestion
Everything between the import and export in the source is **just markdown** using [MDX](https://mdxjs.com/)!
```

export default ({ children }) => <Layout>{children}</Layout>
```

<Caption>
  An example <InlineCode>about.html.mdx</InlineCode> file in a Charge project.
</Caption>

Before we look at deploying our project, let's import the about page we've just created, change your `index.html.jsx` to look like this:

```jsx
import About from './pages/about.html.mdx'

export default () => {
  return (
    <>
      <h1>Welcome to my new Charge site!</h1>
      <About />
    </>
  )
}
```

<Caption>
  An example <InlineCode>index.html.jsx</InlineCode> file in a Charge project.
</Caption>

> Now we have a simple Charge site setup, all that's left to do is deploy it to [Now](/docs/v2). The next section will look at how to configure the [`now.json`](/docs/v2/deployments/configuration) and [`.nowignore`](https://zeit.co/docs/v2/deployments/ignoring-source-paths) files for a quick deployment.

Reviewer: The next section describes itself well enough.

```suggestion
Now that you have a simple Charge site created, you're ready to deploy with [Now](/docs/v2).
```

## Step 3: Deploy Your Charge Project with Now

> Firstly, you need to provide Now with instructions on how to build your Charge project using a [`now.json` configuration file](/docs/v2/deployments/configuration) to be found at the root of our project directory.

Reviewer: ```suggestion
To deploy with Now, you will need to provide a small amount of instructions on how to build your Charge project using a [`now.json` configuration file](/docs/v2/deployments/configuration) at the root of your project directory.
```

Create a [`now.json` file](/docs/v2/deployments/configuration) at the project root and add the following code:

```json
{
  "version": 2,

>   "name": "charge",

Reviewer: ```suggestion
  "name": "my-charge-project",
```

  "builds": [
    {
      "src": "package.json",
      "use": "@now/static-build"
    }
  ],

>   "routes": [{ "src": "/(.*).html", "dest": "/$1" }]

Author: Removed and placed into its own section.

}
```

<Caption>
  An example <InlineCode>now.json</InlineCode> file in a Charge project.
</Caption>

The [`now.json` file](/docs/v2/deployments/configuration) allows you to achieve many things with your [deployment](/docs/v2/deployments/basics/), below is a description of what each property does:

- [`version`](/docs/v2/deployments/configuration/#version) ensures you are using the latest [Now 2.0 platform](/docs/v2/platform/overview) version
- [`name`](/docs/v2/deployments/configuration/#name) defines a project name your deployment will be known by under Now
- [`builds`](/docs/v2/deployments/configuration/#builds) provides one build step using [`@now/static-build`](/docs/v2/deployments/official-builders/static-build-now-static-build/) to instruct Now to statically build the project and deploy the `dist` director, using the `package.json` file as an entrypoint

> - [`routes`](/docs/v2/deployments/configuration/#routes) removes the need to display the `.html` extension in the browser by rewriting paths

Reviewer: Should be in its own section, with the routes property itself. See the caching sections of other guides.

Author: Removed and placed into its own section.

  <Note>
    By default, Now will cache static assets in every{' '}
    <GenericLink href="/docs/v2/platform/regions-and-providers">
      region
    </GenericLink>
    , making your site fast to load wherever you are in the world.
  </Note>

> Next, we will add a `now-build` script to `package.json`. This will be used by the [builder](/docs/v2/deployments/builders/overview/), [`@now/static-build`](/docs/v2/deployments/official-builders/static-build-now-static-build/), to specify what command Now should run to build the site in the cloud. Add this script to `package.json`:

Reviewer: ```suggestion
Lastly, add a `now-build` script to `package.json` which will be used by the [Builder](/docs/v2/deployments/builders/overview/), [`@now/static-build`](/docs/v2/deployments/official-builders/static-build-now-static-build/), to specify what command Now should run to build the site in the cloud:
```

```json
{
  "scripts": {
    ...
    "now-build": "yarn build"
  }
}
```

<Caption>
  Adding a <InlineCode>now-build</InlineCode> script to the{' '}
  <InlineCode>package.json</InlineCode> file in a Charge project.
</Caption>

> At this point we are nearly ready to deploy the project with Now, however, before we do this, create a [`.nowignore` file](https://zeit.co/docs/v2/deployments/ignoring-source-paths) at the root of the project.

Reviewer: ```suggestion
Additionally, to speed up the deployment process by not uploading un-necessary files, create a [`.nowignore` file](https://zeit.co/docs/v2/deployments/ignoring-source-paths) at the root of the project:
```

> The [`.nowignore` file](https://zeit.co/docs/v2/deployments/ignoring-source-paths) is used to [prevent uploading source paths](https://zeit.co/guides/prevent-uploading-sourcepaths-with-nowignore/) to Now, this can dramatically decrease deployment times. Deployment times are decreased as it allows us to prevent build directories and dependencies, such as those found in `/node_modules`, from being uploaded. These are not required as they will be installed again on the server.

Reviewer: I think the above review comment makes this paragraph un-needed, since it's quite long (in fact seems like the longest comment of the guide on one of the least necessary parts)

Author: I have removed this paragraph.

> There are three items you should add to the [`.nowignore` file](https://zeit.co/docs/v2/deployments/ignoring-source-paths); `/node_modules`, `/dist` and`/tmp` - the directory Charge creates when serving content locally:

Reviewer: ```suggestion
There are three items, in this example, that you should add to the [`.nowignore` file](https://zeit.co/docs/v2/deployments/ignoring-source-paths); `/node_modules`, `/dist`, and`/tmp` - the directory Charge creates when serving content locally:
```

```
node_modules
tmp
dist
```

<Caption>
  An example <InlineCode>.nowignore</InlineCode> file in a Charge project.
</Caption>

> We are now ready to deploy our application.

Reviewer: ```suggestion
Finally, deploy the application with [Now](/docs/v2/getting-started/introduction-to-now).
```

If you have not yet installed Now, you can do so by installing the [Now Desktop app](/docs/v2/getting-started/installation/#now-desktop) which installs Now CLI automatically, or by [installing Now CLI](/docs/v2/getting-started/installation/#now-cli) directly.

> Now allows you to deploy your project from the terminal with just one command, go ahead and do so:

Reviewer: ```suggestion
Now allows you to [deploy your project](/docs/v2/deployments/basics) from the terminal with just one command:
```

<TerminalInput>now</TerminalInput>
<Caption>
  Deploying an application with Now using only one command.
</Caption>

> You will see a short build step in your terminal followed by the news that your Charge project has now been [deployed](https://zeit.co/docs/v2/deployments/basics/), congratulations!

Reviewer: Can we include what the other guides include, a deployment URL? It should be something like `my-charge-project-msweeneydev.now-examples.now.sh`, saying that this is what you'll get from this example, with a link to the aliasing document for more information.
